remix/app/routes/_app.newsletter.update.$id.test.tsx

```diff
@@ -35,65 +36,112 @@ describe('PUT /newsletter/update/:id', () => {
     );
 
     describe('given valid newsletter params', () => {
-      it('update a newsletter', async () => {
-        const updateResultMock = { count: 1 };
+      it('returns a newsletter', async () => {
         const newsletterAttributes = { userId: '1' };
         const newsletter = { ...newsletterFactory, ...newsletterAttributes };
 
-        prismaMock.newsletter.findUnique.mockResolvedValue(newsletter);
-        prismaMock.newsletter.updateMany.mockResolvedValue(updateResultMock);
-
-        const body = new URLSearchParams({
-          name: newsletter.name,
-          content: newsletter.content,
-        });
+        prismaMock.newsletter.findFirst.mockResolvedValue(newsletter);
 
         const request = makeRequest({
           url: `/newsletter/update/${newsletter.id}`,
-          method: 'put',
-          body,
+          method: 'get',
         });
 
-        const result: any = await action({
+        const result: any = await loader({
           request,
           params: {},
           context: {},
         });
 
-        expect(await result.json()).toMatchObject({
+        expect(await result.json()).toMatchObject<Newsletter>({
+          id: newsletter.id,
           name: expect.any(String),
           content: expect.any(String),
+          userId: newsletter.userId,
+          createAt: expect.any(String),
+          updateAt: expect.any(String),
         });
       });
     });
+  });
 
-    describe('given an invalid newsletter params', () => {
-      it('returns error NOT FOUND', async () => {
-        const newsletter = { ...newsletterFactory };
+  describe('action', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
```

Comment:
Shouldn't this moved to `afterEach` (like a teardown) in the first `describe` block?

```
describe('Newsletter update', () => {
   afterEach(() => {
      jest.clearAllMocks();
  });

  describe('loader', () => {
}
```

Reply:
Fixed here: [b357dd3](https://github.com/nimblehq/js-framework-benchmark/pull/130/commits/b357dd39fc2364fe1acf62523fa9f7463e88caed)